binary_search_tree/binary_search_tree.py

```diff
@@ -5,35 +5,35 @@ def __init__(self, value):
     self.right = None
 
   def insert(self, value):
-    print('inserting:', value)
-    if value >= self.value:
-      if self.right == None:
-        self.right = BinarySearchTree(value)
+    # print('inserting:', value)
+    
+    if (value < self.value):
+      if (self.left):
+        self.left.insert(value)
       else:
-        self.right.insert(value)
-    else:
-      if self.left == None:
         self.left = BinarySearchTree(value)
+    else:
+      if (self.right):
+        self.right.insert(value)
       else:
-        self.left.insert(value)
+        self.right = BinarySearchTree(value)
 
   def contains(self, target):
```

Comment:
looks great too! very clean, nice use of recursion (and the fact that a BST can be thought of multiple little BSTs connected to one another, which lets you use recursion so smoothly)